mava/evaluator.py

```diff
@@ -93,7 +93,10 @@ def not_done(carry: Tuple) -> bool:
         }
         # Log won episode if win rate is required.
         if log_win_rate:
-            eval_metrics["won_episode"] = jnp.all(final_state.timestep.reward >= 1.0).astype(int)
+            if "won_episode" in final_state.timestep.extras:
+                eval_metrics["won_episode"] = final_state.timestep.extras["won_episode"]
+            else:
+                eval_metrics["won_episode"] = int(jnp.all(final_state.timestep.reward >= 1.0))
```

Comment:
Not a fan of the `if` here, but I much prefer adding `won_episode` to the extras! Can you add this inside the `SmaxWrapper` please and then remove this if and all check the extras?

Reply:
done, there was also a small oversight in the `evaluator.py` l 287 that caused the `log_win_rate` to always be false since the `env_name` was changed to "Smax"
`log_win_rate = config.env.env_name in ["HeuristicEnemySMAX", "LearnedPolicyEnemySMAX"]`